File: emailsender/languages/go/relay_policy.go
package emailsender

import "fmt"

type RelayPolicy struct { Authorized, OpenRelay, SpoofFrom, PrivacyMode bool }

func (p RelayPolicy) Validate() error {
    if p.OpenRelay || p.SpoofFrom { return fmt.Errorf("open relays and From spoofing are disabled") }
    if !p.Authorized { return fmt.Errorf("relay authorization is required") }
    return nil
}

File: emailsender/languages/typescript/relay-policy.ts
export interface RelayPolicy { authorized: boolean; openRelay: boolean; spoofFrom: boolean; privacyMode: boolean; }

export function validateRelay(policy: RelayPolicy): void {
  if (policy.openRelay || policy.spoofFrom) throw new Error('Open relays and From spoofing are disabled');
  if (!policy.authorized) throw new Error('Relay authorization is required');
}
